newIDE/app/src/VariablesList/index.js

```diff
@@ -173,16 +192,29 @@ export default class VariablesList extends React.Component<Props, State> {
         variable={variable}
         disabled={depth !== 0}
         depth={depth}
+        variableMetadata={variableMetadata}
         errorText={
           this.state.nameErrors[variable.ptr]
             ? 'This name is already taken'
             : undefined
         }
         onChangeValue={text => {
-          variable.setString(text);
+          // if it's inherited and value is different from default, create an instance variable
+          if (text !== defaultValue && !variablesContainer.has(name)) {
```

Comment:
Do we need this check? 
I would say that as soon as an inherited variable is edited, boom, you create an instance variable for it. No need to check if it has a default value, because we don't care :) 
And no need to check if variablesContainer has it. Why? Because as soon as you create the instance variable, the instance variable will "hide" the inherited variable.

Reply:
Unfortunately we do, because _createInheritedInstanceVariable actually creates it and adds it

Reply:
I could move the check to the function if you want to?

Reply:
I agree that "_createInheritedInstanceVariable actually creates it and adds it". But at this moment, the instance variable will hide the object variable, so nothing will be created twice.

Correct?
We agree that if an instance variable with name "Variable1" is created, then object variable with name "Variable1" will be hidden right? So in this case, you don't care about default value.
React will automatically re-render your component, and your logic will make so that "Variable1" is now the instance variable.

Reply:
(if React does not update the component, call this.forceUpdate()).
Please verify that as soon as you create an instance variable, then the object variable is not shown anymore. In this case, you don't need to check default value.
If that's not the case (i.e: if you prefer to still display object variables that are overwritten by instance variables), then you can just have `!variablesContainer.has(name)` as condition and forget about default value. Default value should not change anything.

Reply:
I thought the same, but then I saw that it actually created it twice, maybe I need to change something else in the code

Reply:
Ok, mmm in this case make sure to call this.forceUpdate(); after you've created the variable. You can still keep `!variablesContainer.has(name)` for safety then

Reply:
But again, `text !== defaultValue` is not needed right?

Reply:
not any more